Update README.md to make it easier to understand for novices. (#129)
eWaterCycle/infra: README.md

# Instructions for system administrators to deploy the eWaterCycle platform

![Ansible Lint](https://github.com/eWaterCycle/infra/workflows/Ansible%20Lint/badge.svg)
[![Concept DOI](https://zenodo.org/badge/DOI/10.5281/zenodo.1462548.svg)](https://doi.org/10.5281/zenodo.1462548)

This repo contains (codified) instructions for deploying the eWaterCycle platform. The target audience of these instructions are system administrators. For more information on the eWaterCycle platform (and how to deloy it) see the [eWaterCycle documentation[(https://ewatercycle.readthedocs.io/).

For instructions on how to use the machine as deployed by this repo see the [User guide](USER.md).

These instructions assume you have some basic knowledge of [vagrant](https://vagrantup.com) and
 [Ansible](https://docs.ansible.com/ansible/latest/index.html).

## Setup of eWaterCycle platform on the SURF Research cloud

> De environment used by the eWaterCycle platform development team is the [SURF Research Cloud](https://servicedesk.surfsara.nl/wiki/display/WIKI/Research+Cloud+Documentation). The setup instructions in this repo will create an eWatercycle application(a sort-of VM template) that when started will create a machine with:

Reviewer: typo (de instead of the)

Author: Fixed

* Explorer: web visualization of available models / parameter sets combinations and a way to generate Jupyter notebooks
* Jupyter Hub: to interactivly generate forcings and perform experiments on hydrological models using the [eWatercycle Python package](https://ewatercycle.readthedocs.io/)
* ERA5 and ERA-Interim global climate data, which can be used to generate forcings
* Installed models and their example parameter sets

> 

Reviewer: Maybe explain briefly that the one setting up the machine needs to have research budget, but can distribute access to the machine to others (using SRAM)?

Author: Added a note.

# Instructions for system administrators to deploy the eWaterCycle platform

![Ansible Lint](https://github.com/eWaterCycle/infra/workflows/Ansible%20Lint/badge.svg)
[![Concept DOI](https://zenodo.org/badge/DOI/10.5281/zenodo.1462548.svg)](https://doi.org/10.5281/zenodo.1462548)

> This repo contains (codified) instructions for deploying the eWaterCycle platform. The target audience of these instructions are system administrators. For more information on the eWaterCycle platform (and how to deloy it) see the [eWaterCycle documentation[(https://ewatercycle.readthedocs.io/).

Reviewer 2: ```suggestion
This repo contains (codified) instructions for deploying the eWaterCycle platform. The target audience of these instructions are system administrators. For more information on the eWaterCycle platform (and how to deploy it) see the [eWaterCycle documentation[(https://ewatercycle.readthedocs.io/).
```

For instructions on how to use the machine as deployed by this repo see the [User guide](USER.md).

These instructions assume you have some basic knowledge of [vagrant](https://vagrantup.com) and
 [Ansible](https://docs.ansible.com/ansible/latest/index.html).

## Setup of eWaterCycle platform on the SURF Research cloud

The hardware environment used by the eWaterCycle platform development team is the [SURF Research Cloud](https://servicedesk.surfsara.nl/wiki/display/WIKI/Research+Cloud+Documentation). Starting a machine on the Surf Research Cloud requires that you have research budget with SURF, for more info see the website of [SURF](https://www.surf.nl/en/research-it/apply-for-access-to-compute-services). Once running, access to the machine can be shared to anyone.

> The setup instructions in this repo will create an eWatercycle application(a sort-of VM template) that when started will create a machine with:

Reviewer 2: ```suggestion
The setup instructions in this repo will create an eWaterCycle application (a sort-of VM template) that when started will create a machine with:
```

* Explorer: web visualization of available models / parameter sets combinations and a way to generate Jupyter notebooks
* Jupyter Hub: to interactivly generate forcings and perform experiments on hydrological models using the [eWatercycle Python package](https://ewatercycle.readthedocs.io/)
* ERA5 and ERA-Interim global climate data, which can be used to generate forcings
* Installed models and their example parameter sets

An application on the SURF Research cloud is provisioned by running an Ansible playbook (research-cloud-plugin.yml).

In addition to the standard VM storage, additional read-only datasets are mounted at `/mnt/data` from dCache using rclone. They may contain things like:

* climate data, see <https://ewatercycle.readthedocs.io/en/latest/system_setup.html#download-climate-data>
* observation
* parameter-sets
* singularity-images of hydrological models wrapped in grpc4bmi servers

Previously the eWatercycle platform consisted of multiple VM on SURF HPC cloud, see [v0.1.2 release](https://github.com/eWaterCycle/infra/releases/tag/v0.1.2) for that code.

## Setup of eWaterCycle platform on a local test VM

> Deplying a local test VM is mostly useful for developing the SURF Research Cloid applications. This vagrant setup should work on any Linux or Windows machine with sufficient memory (8Gb, mostly depends on the models you run), cpu (More is better, one core will do if needed), and storage (At least 200Gb) available.

Reviewer 2: ```suggestion
Deploying a local test VM is mostly useful for developing the SURF Research Cloud applications. This vagrant setup creates a virtual machine with 8Gb memory, 4 virtual cores, and 70Gb storage. This should work on any Linux or Windows machine.
```

> * parameter-sets, example use cases and bigger ones

Reviewer: bigger ones is unclear, remove?

Author: Removed

* singularity-images of hydrological models wrapped in grpc4bmi servers

Previously the eWatercycle platform consisted of multiple VM on SURF HPC cloud, see [v0.1.2 release](https://github.com/eWaterCycle/infra/releases/tag/v0.1.2) for that code.

## Setup of eWaterCycle platform on a local test VM

> Deplying a local test VM is mostly useful for developing the SURF Research Cloid applications. This vagrant setup should work on any Linux or Windows machine with sufficient memory (8Gb, mostly depends on the models you run), cpu (More is better, one core will do if needed), and storage (At least 200Gb) available.

Author: Can someone verify these numbers?

Reviewer: verified. These numbers make sense.
